DogServiceExample/src/main/java/com/common/rest/DogService.java

```diff
@@ -0,0 +1,110 @@
+package com.common.rest;
+
+import com.common.json.JsonHelper;
+import com.common.app.Dog;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.HashMap;
+import java.util.Map;
+
+
+@Path("/")
+public class DogService {
```

Comment:
Here you convert method parameters from Json and responses to Json manually. There mush be a way to do this automatically: you can define function like `public Dog createNewDog(Dog dog)` and the parameters and responses will be converted by the application server.